packages/storage/amplify_storage_s3_dart/test/storage_s3_service/transfer/database_html_test.dart

```diff
@@ -0,0 +1,64 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+@TestOn('browser')
+import 'dart:html';
+
+import 'package:amplify_storage_s3_dart/src/storage_s3_service/transfer/database/database_html.dart';
+import 'package:amplify_storage_s3_dart/src/storage_s3_service/transfer/database/transfer_record.dart';
+import 'package:test/test.dart';
+
+void main() {
+  group('TransferDatabase for web', () {
+    const testUploadId = 'test-upload-Id';
+    const testObjectKey = 'test-object-Key';
+    final testCreatedAt = DateTime(2022, 1, 1).toIso8601String();
+
+    final testTransferRecord = TransferRecord(
+      uploadId: testUploadId,
+      objectKey: testObjectKey,
+      createdAt: testCreatedAt,
+    );
+
+    final testTransferRecordJsonString = testTransferRecord.toJsonString();
+
+    late TransferDatabase transferDatabase;
+
+    setUp(() {
+      transferDatabase = TransferDatabase();
+      window.localStorage.clear();
+      window.localStorage['${localStorageKeyPrefix}test-uuid'] =
+          testTransferRecordJsonString;
+    });
+
+    test('insertTransferRecord to local storage', () async {
+      final recordId =
+          await transferDatabase.insertTransferRecord(testTransferRecord);
+      expect(recordId.startsWith(localStorageKeyPrefix), isTrue);
```

Comment:
Looking at the impl. of `insertTransferRecord` on Web, the returned `recordId` wasn't actually read from the storage, hence, I think this test can be improved a bit -

In addition to verify the `recordId`, after inserting, we can read it from the storage and compare the result.